android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java

```diff
@@ -0,0 +1,736 @@
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.SwitchCompat;
+import androidx.core.content.res.ResourcesCompat;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.google.android.material.bottomsheet.BottomSheetBehavior;
+import com.google.android.material.bottomsheet.BottomSheetDialog;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+
+import org.json.JSONException;
+
+import org.chromium.base.Log;
+import org.chromium.brave_rewards.mojom.PublisherStatus;
+import org.chromium.brave_rewards.mojom.WalletStatus;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.BraveRewardsBalance;
+import org.chromium.chrome.browser.BraveRewardsExternalWallet;
+import org.chromium.chrome.browser.BraveRewardsHelper;
+import org.chromium.chrome.browser.BraveRewardsNativeWorker;
+import org.chromium.chrome.browser.BraveRewardsObserver;
+import org.chromium.chrome.browser.BraveWalletProvider;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.customtabs.CustomTabActivity;
+import org.chromium.chrome.browser.util.TabUtils;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.Locale;
+
+public class RewardsTippingPanelFragment
+        extends BottomSheetDialogFragment implements BraveRewardsObserver {
+    final public static String TAG_FRAGMENT = "tipping_panel_tag";
+    private static final String TAG = "TippingPanelFragment";
+
+    private static final String WEB3_URL = "web3_url";
+    private BraveRewardsNativeWorker mBraveRewardsNativeWorker;
+    private String walletType = BraveRewardsNativeWorker.getInstance().getExternalWalletType();
+
+    private TextView radio_tip_amount[] = new TextView[4];
+    private double[] mTipChoices;
+    private TextView currency1TextView;
+    private TextView currency2TextView;
+    private EditText currency1ValueEditTextView;
+    private TextView currency1ValueTextView;
+    private TextView currency2ValueTextView;
+    private boolean toggle = true;
+    public static final double AMOUNT_STEP_BY = 0.25;
+    public static final int MAX_BAT_VALUE = 100;
+    private static final double DEFAULT_AMOUNT = 0.0;
+
+    private static final int DEFAULT_VALUE_OPTION_1 = 1;
+    private static final int DEFAULT_VALUE_OPTION_2 = 5;
+    private static final int DEFAULT_VALUE_OPTION_3 = 10;
+
+    private View mContentView;
+    private Button mSendButton;
+    private Button mWeb3WalletButton;
+    private int mCurrentTabId = -1;
+    private double mBalance;
+    private String mWeb3Url;
+
+    private boolean mIsLogoutState;
+    private double mAmountSelected;
+
+    private double rate;
+    private boolean mIsBatCurrency;
+    private BraveRewardsExternalWallet mExternalWallet;
+    private ProgressBar mTipProgressBar;
+
+    private TextView usdSymbol1;
+    private TextView usdSymbol2;
```

Comment:
could you please make all fields names consistent across that new file? `usdSymbol2` --> `mUsdSymbol2`, `rate` --> `mRate` and etc. As it's very hard to understand is that var a member or a local var.

Reply:
updated

Reply:
Please update all members in that file
```
    private TextView currency1TextView;
    private TextView currency2TextView;
    private EditText currency1ValueEditTextView;
    private TextView currency1ValueTextView;
    private TextView currency2ValueTextView;
```

Reply:
updated